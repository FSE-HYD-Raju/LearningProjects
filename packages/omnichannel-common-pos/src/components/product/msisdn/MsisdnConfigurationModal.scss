@import '../../../variables.scss';

.MsisdnConfigurationModal {
	display: flex;
	align-items: flex-start;

	&-modal-container {
		display: flex;
	}
	&-pattern-selector {
		flex: 2;
		height: fit-content;
	}
	&-price-component {
		flex: 2;
		margin-left: 8rem;
		border-bottom: $border-bottom;
		box-shadow: $box-shadow;
		height: fit-content;
	}
	&-button-wrapper {
		padding: 20px;
		background: $background-color;
		border-top: $border-top;
	}
    @include media-breakpoint-between(xs, sm) {
		&-modal-container {
			display: block;
		}
		&-pattern-selector {
			flex: 4;
		}
		&-price-component {
			flex: 4;
			margin-left: unset;
		}
	}
}