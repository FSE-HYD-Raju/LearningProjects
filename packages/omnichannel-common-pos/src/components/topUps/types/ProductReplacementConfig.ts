interface ProductReplacementConfig {
	individualId?: string;
	agreementId?: string;
	productId?: string;
	productOfferingId?: string;
	replaceType?: string;
	configuration?: object;
}
export { ProductReplacementConfig };
