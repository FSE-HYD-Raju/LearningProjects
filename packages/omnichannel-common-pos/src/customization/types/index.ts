import { CustomizationBootstrapType, GetStoreConfig } from "./CustomizationBootstrapType";
import { CustomizationClass } from "./CustomizationClass";
import { CustomizationFunction } from "./CustomizationFunction";
import { CustomizationPointsMapping } from "./CustomizationPointsMapping";

export {
	GetStoreConfig,
	CustomizationBootstrapType,
	CustomizationClass,
	CustomizationFunction,
	CustomizationPointsMapping
};
