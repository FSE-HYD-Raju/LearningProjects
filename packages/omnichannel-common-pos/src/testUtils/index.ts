import ReactWidgetsTestUtils from "./ReactWidgetsTestUtils";
import * as MockDataMaker from "./mock";
const {
	shallowWithContextAndRouterProps,
	shallowWithContextAndLifecycle,
	deepShallowWithContextAndLifecycle,
	deepShallowWithContext,
	shallowWithContext,
	mountWithContextAndRouterProps,
	mountWithContext,
	attachWithContext,
	detach
} = require("./custom-mounters");
const TestUtils = require("./TestUtils");
const MockData = require("./MockData");
const SimpleDataMock = require("./mock/simpleDataMock");

export {
	MockData,
	ReactWidgetsTestUtils,
	TestUtils,
	SimpleDataMock,
	shallowWithContextAndRouterProps,
	shallowWithContextAndLifecycle,
	deepShallowWithContextAndLifecycle,
	deepShallowWithContext,
	shallowWithContext,
	mountWithContextAndRouterProps,
	mountWithContext,
	attachWithContext,
	detach,
	MockDataMaker
};

export * from "./MockRouterProps";

export default TestUtils;
