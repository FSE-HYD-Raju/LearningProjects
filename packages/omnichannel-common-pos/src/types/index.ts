import { ContextType, contextTypesValidationMap } from "./ContextType";

export {
	ContextType,
	contextTypesValidationMap
};
