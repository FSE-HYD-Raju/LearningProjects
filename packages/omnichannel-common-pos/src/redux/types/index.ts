import { ValidatePortInType } from "../../components/product/msisdn/MsisdnPortInSelector";
import { ConnectFluxReduxChannelType } from "./ConnectFluxReduxChannelType";
import { Disposers } from "./Disposers";
import { FluxAlt } from "./FluxAlt";
import { FluxReduxMap } from "./FluxReduxMap";
import { FluxStore } from "./FluxStore";
import { ReduxFluxActionMigration } from "./ReduxFluxActionMigration";
import { FluxWithActions } from "./fluxActions/FluxWithActions";

export {
	ConnectFluxReduxChannelType,
	Disposers,
	FluxAlt,
	FluxReduxMap,
	FluxStore,
	ReduxFluxActionMigration,
	FluxWithActions,
	ValidatePortInType
};

export * from "./utils/Omit";
export * from "./utils/HocResult";
export * from "./fluxActions";
export * from "../models/cms/cms.types";
export * from "../models/auth/auth.types";
export { CmsState } from "../models/cms/cms.reducers";
export { CmsAdminState } from "../models/cmsAdmin/cmsAdmin.reducers";
export * from "../models/auth/auth.types";
export * from "../models/service/service.types";
export * from "../models/support/support.types";
export * from "../models/comparison/comparison.types";
export * from "../models/consul/consul.types";
export * from "../models/currency/currency.types";
export * from "../models/eCare/changeSim/changeSim.types";
export * from "../models/delivery/delivery.types";
export * from "../models/digitalLife/digitalLife.types";
export * from "../models/eCare/activateSim/activateSim.types";
export * from "../models/eCare/suspension/suspension.types";
export * from "../models/feature/feature.types";
export * from "../models/lifecycle/lifecycle.types";
export * from "../models/organization/organization.types";
export * from "../models/productOfferingConfiguration/productOfferingConfiguration.types";
export * from "../models/schema/schema.types";
export * from "../models/location/location.types";
export * from "../models/eligibility/eligibility.types";
export * from "../models/productOfferings/productOfferings.types";
export * from "../models/user/user.types";
export * from "../models/basket/basket.types";
export * from "../models/customer/customer.types";
export * from "../models/sales/sales.types";
export * from "./SupportCases";
export * from "../models/versionInformation/versionInformation.types";
export * from "../models/msisdnSelection/msisdnSelection.types";
export * from "../models/portIn/portIn.types";
export * from "../models/marketingConsent/marketingConsent.types";
export * from "../models/category/category.types";
export * from "../models/b2cCheckout/b2cCheckout.types";
export { ProvinceAndCityState, Province, CityType } from "../models/provinceAndCity/provinceAndCity.types";

export * from "./routes/HasPath";
export * from "./routes/ApplicationRoute";
export * from "./HasSchema";
export * from "./HasFlux";
export * from "./HasId";
export * from "./HasPrices";
export * from "./BillingInterval";
export * from "./Cardinality";
export * from "./Characteristic";
export * from "./CharacteristicValue";
export * from "./CommercialEnrichments";
export * from "./Country";
export * from "./Links";
export * from "./FilterLabel";
export * from "./Gender";
export * from "./ImageType";
export * from "./Msisdn";
export * from "./MsisdnConfig";
export * from "./Price";
export * from "./SimplePrice";
export * from "./PriceRange";
export * from "./PriceType";
export * from "./PriceRangeBoundaries";
export * from "./ProductFilter";
export * from "./ProductOffering";
export * from "./ProductOfferingGroup";
export * from "./AlternateProductOffering";
export * from "./StockLevel";
export * from "./ProductTypes";
export * from "./ServiceUnit";
export * from "./Resource";
export * from "./SimCard";
export * from "./UnitOfMeasure";
export * from "./Allowance";
export * from "./ValidityPeriod";
export * from "./ValueRegulator";
export * from "./AutoCharacteristicConfigurationType";
export * from "./ChargingBalances";
export * from "./ChargingBalanceChanges";
export * from "./ProductsModifyConfig";
export * from "./Product";
export * from "./Agreement";
export * from "./ContactMediaType";
export * from "./EmailAddress";
export * from "./PhoneNumber";
export * from "./PostalAddress";
export * from "./Location";
export * from "./StreetAddressTokens";
export * from "./IssuingAuthority";
export * from "./Identification";
export * from "./CommonPerson";
export * from "./RelatedPerson";
export * from "./OrderLifeCycleStatus";
export * from "./Order";
export * from "./OrderCancel";
export * from "./OrderItem";
export * from "./OrderTotalsType";
export * from "./Person";
export * from "./MsisdnLifecycleStatus";
export * from "./Basket";
export * from "./payment/ContextualPayment";
export * from "./payment/ContextualPaymentMethod";
export * from "./payment/ContextualPaymentValidation";
export * from "./payment/CustomerPaymentMethod";
export * from "./payment/CustomerPaymentMethodType";
export * from "./payment/PaymentMethodsFiltersType";
export * from "./payment/CreditCard";
export * from "./payment/Role";
export * from "./BasketItem";
export * from "./SpecSubType";
export * from "./SpecType";
export * from "./TargetLifecycleStatus";
export * from "./BasketLifecycleStatus";
export * from "./BasketValidationAction";
export * from "./BasketValidationInformation";
export * from "./BasketProduct";
export * from "./PersonToPersonRelationship";
export * from "./CustomerCase";
export * from "./Category";
export * from "./SubCategory";
export * from "./ProductLifecycleStatus";
export * from "./UsageLimits";
export * from "./UsageCounters";
export * from "./Identifier";
export * from "./Notification";
export * from "./Document";
export * from "./BillingBalance";
export * from "./ShippingMethod";
export * from "./ChangeResourceInitialization";
export * from "./ChangeResource";
export * from "./Reason";
export * from "./MsisdnReservation";
export * from "./payment/PaymentStatus";
export * from "./payment/PaymentInvocation";
export * from "./Owner";
export * from "./CustomerInteraction";
export * from "./Initiator";
export * from "./InteractionDetails";
export * from "./AttachedDocument";
export * from "./Discount";
export * from "./Interval";
export * from "./BasketPaymentReceipt";
export * from "./PaginationData";
export * from "./PersonsOrderData";
export * from "./PersonsOrderState";
export * from "./OrderStatusFilter";
export * from "./SalesInfo";
export * from "./Shipment";
export * from "./AxiosResponse";
export * from "./CommercialEnrichmentName";
export * from "./UsageEvent";
export * from "./UsagePrice";
export * from "./NonMonetaryAmount";
export * from "./EventDetails";
export * from "./LifecycleChangeAction";
export * from "./Service";
export * from "./ServiceLifeCycleStatus";
export * from "./Specification";
export * from "./CustomerAccount";
export * from "./CustomerAccountLifecycleStatus";
export * from "./RecurringTopUpModelType";
export * from "../../components/priceComponent/model/PriceComponentPriceType";
export * from "./ErrorType";
export * from "./PostalAddressValidationErrorType";
export * from "./ProductModification";
export * from "./CallForwardingServiceModify";
export * from "./ProductOfferingFilter";
export * from "./ProductOfferingFilterMatcher";
export * from "./PortInDecision";
export * from "./ProductOfferingFilter";
export * from "./ProductOfferingFilterMatcher";
export * from "./DomainContext";
export * from "./ServiceModification";
export * from "./ResourceStocks";
export * from "./ResourceInventories";
export * from "./Inventory";
export * from "./SalesOrganizationRole";
export * from "./NominationCharacteristics";
export * from "./NominationPOCharacteristicsConfig";
export * from "./BasketItemsColumn";
export * from "./OrderItemsColumn";
export * from "./OrdersColumn";
export * from "./BasketsColumn";
export * from "./ConsumptionData";
export * from "./ProductsReplaceType";
export * from "./OmnichannelApiResponse";
export * from "./ProductConfiguration";
export * from "./TopUpCharacteristic";
export * from "./RecurringTopUpIntervalType";
export * from "./RecurringTopUpCharacteristicsValue";
export * from "./RecurringTopUpProductType";
export * from "./HasProduct";
export * from "./Store";
export * from "./QueryStatus";
export * from "./DeviceReservation";
export * from "./HasIntl";
export * from "./ProductUsageColumn";
export * from "./WorkforceAppointment";
