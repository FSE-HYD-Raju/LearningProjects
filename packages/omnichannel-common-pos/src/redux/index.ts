"use strict";

import { default as actions, AppActions } from "./actions";
import { default as reducers, AppState } from "./reducers";
import { InterceptReducers } from "./reducers/reducerUtils";

import { default as DevTools } from "./DevTools";

import { connectFluxRedux, getDefaultFluxStoreMap } from "./connectFluxRedux";
import { configureStore, AppStore, StoreConfig } from "./store";
import * as actionTypes from "./actions";
import * as decorators from "./decorators";

import * as channelUtils from "./utils/channel";
export * from "./utils";
import * as Selectors from "./selectors";

import { initialState } from "./reducers/initialState";
import BaseService from "./services/BaseService";
export * from "./settings/core";
import { productToBasketItem as productToBasketItemTypes } from "./models/feature/feature.types";
// uses "configureStore" from ./store, should be imported after ./store
import * as ReduxTestUtils from "./utils/ReduxTestUtils";

export {
	decorators,
	Selectors,
	actions,
	reducers,
	DevTools,
	StoreConfig,
	InterceptReducers,
	AppStore,
	initialState,
	AppActions,
	AppState,
	connectFluxRedux,
	configureStore,
	getDefaultFluxStoreMap,
	ReduxTestUtils,
	channelUtils,
	actionTypes,
	BaseService,
	productToBasketItemTypes,
};

export * from "./actions/actionsPayload";
export * from "./services/Rest";
export * from "./services";
export * from "./types";
export * from "./decorators";
export * from "./models/msisdnSelection/msisdnSelection.actions";
export * from "./models/productOfferingConfiguration/productOfferingConfiguration.utils";
