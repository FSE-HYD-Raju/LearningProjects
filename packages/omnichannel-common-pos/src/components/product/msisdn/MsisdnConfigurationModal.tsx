import { PureComponent } from "react";
import cssns from "../../../utils/cssnsConfig";
import { get } from "lodash";
import ProductOfferingUtil from "../../../utils/ProductOfferingUtil";
import {
	ProductOffering,
	ProductPath,
	BasketActionAddProductToBasketOptions,
	MsisdnWithCost,
} from "../../../redux";
import MsisdnSelectionContainer from "../../msisdnSelection/MsisdnSelectionContainer";
import { findMsisdnGroupFromProductOffering } from "../../../redux/models/msisdnSelection/msisdnSelection.utils";
import ComplexPriceComponent from "../../priceComponent/ComplexPriceComponent";
import { Configurations } from "../../../redux";

const { React } = cssns("MsisdnConfigurationModal");

interface MsisdnConfigurationModalActionProps {
	actions: {
		handleClose: () => void;
		selectProductOffering: (path: ProductPath, value: string, productOfferings: ProductOffering[]) => void;
		saveNumber: (path: ProductPath, key: string, value: string) => void;
		addProduct: (options: BasketActionAddProductToBasketOptions) => void;
	};
}

interface MsisdnConfigurationModalStateProps {
	product: ProductOffering;
	path: ProductPath;
	msisdnConfig: { msisdn?: string };
	userOpened?: boolean;
	selectedMsisdn?: MsisdnWithCost;
	priceAttribute: string;
	configurations: Configurations;
}

type MsisdnConfigurationModalProps = MsisdnConfigurationModalActionProps & MsisdnConfigurationModalStateProps;

interface MsisdnConfigurationModalState {}

class MsisdnConfigurationModal extends PureComponent<MsisdnConfigurationModalProps, MsisdnConfigurationModalState> {
	componentWillReceiveProps(newprops: MsisdnConfigurationModalProps) {
		const { product, userOpened } = newprops;

		/* User has tried to add a product to basket, but it required MSISDN configuration. If user has successfully
		configured the product, add the product to basket */
		if (this.props.product.id === product.id && !get(this.props, "msisdnConfig.msisdn")
			&& get(newprops, "msisdnConfig.msisdn") && !userOpened) {
			// addProduct && addProduct();
			// TODO: Support adding directly to basket
		}
	}

	saveNumber = (selectedNumberType: ProductOffering) => {
		const numberPath = ProductOfferingUtil.getPathToProductOfferingById(this.props.product, selectedNumberType.id);

		const pogPath = numberPath && numberPath.slice(0, -1);
		const msisdnPog = findMsisdnGroupFromProductOffering(this.props.product);
		const numberClassPOs = ProductOfferingUtil.getProductOfferings(msisdnPog);

		if (numberPath && pogPath && msisdnPog && this.props.selectedMsisdn && this.props.selectedMsisdn.msisdn) {
			this.props.actions.selectProductOffering(numberPath.slice(0, -1), selectedNumberType.id, numberClassPOs);

			this.props.actions.saveNumber(
				this.props.path || pogPath.slice(0, -1),
				"CH_NumberResource",
				this.props.selectedMsisdn.msisdn
			);
		}
	};

	render() {
		const { product, selectedMsisdn, priceAttribute, configurations } = this.props;
		if (selectedMsisdn && selectedMsisdn.productOffering &&
			product.attributes && product.attributes.productOfferingGroups) {
			const pogs = ProductOfferingUtil.updateProductOfferingGroupsWithSelectedMsisdn(
				ProductOfferingUtil.getProductOfferingGroups(product),
				selectedMsisdn.productOffering
			);

			// Now fill product with updated productOfferingGroups
			product.attributes.productOfferingGroups.splice(0, product.attributes.productOfferingGroups.length, ...pogs);
		}

		return product ? (
			<div className="modal-container">
				<div className="pattern-selector">
					<MsisdnSelectionContainer productId={product.id} />
				</div>
				<div className="price-component">
					<ComplexPriceComponent
						product={product}
						configurations={configurations}
						priceAttribute={priceAttribute}
					/>
				</div>
			</div>
		) : null;
	}
}

export default MsisdnConfigurationModal;
export {
	MsisdnConfigurationModalActionProps,
	MsisdnConfigurationModalStateProps,
	MsisdnConfigurationModalProps,
	MsisdnConfigurationModalState
};
