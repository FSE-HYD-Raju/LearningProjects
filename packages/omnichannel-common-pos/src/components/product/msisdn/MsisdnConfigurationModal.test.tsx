import * as React from "react";
import { shallowWithContext } from "../../../testUtils";
import { ProductOffering } from "../../../redux/types";
import MsisdnConfigurationModal from "./MsisdnConfigurationModal";

describe("MsisdnConfigurationModal", () => {
	const props = {
		actions: {
			addProduct: () => {},
			saveNumber: () => {},
			reserveMsisdn: () => {},
			handleClose: () => {},
			selectProductOffering: () => {},
		},
		priceAttribute: "taxFreeAmount",
		configurations: {},
		msisdnConfig: {},
		reservationAttributes: {
			msisdns: [
				1234567, 1232323, 2132132
			]
		},
		path: [],
		product: {
			id: "testi"
		} as ProductOffering
	};

	it("succeeds at shallow mount without props", () => {
		const wrapper = shallowWithContext(
			<MsisdnConfigurationModal {...props} />
		);
		expect(wrapper).toMatchSnapshot();
	});
});
