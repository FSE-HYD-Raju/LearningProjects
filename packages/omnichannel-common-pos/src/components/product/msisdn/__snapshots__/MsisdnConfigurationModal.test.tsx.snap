// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MsisdnConfigurationModal succeeds at shallow mount without props 1`] = `
<div
  className="MsisdnConfigurationModal-modal-container"
>
  <div
    className="MsisdnConfigurationModal-pattern-selector"
  >
    <Connect(MsisdnSelection)
      productId="testi"
    />
  </div>
  <div
    className="MsisdnConfigurationModal-price-component"
  >
    <ComplexPriceComponent
      configurations={Object {}}
      priceAttribute="taxFreeAmount"
      product={
        Object {
          "id": "testi",
        }
      }
    />
  </div>
</div>
`;
